### PR TITLE
Add support for VLAN ID in Tinkerbell provider:

### PR DESCRIPTION
**Pull request by jacobweinstock**
*Issue #, if available:*
#3596

*Description of changes:*
This enables trunked port environments to work with EKSA provider Tinkerbell. Official docs to follow. This functionality will only work properly with updated Tinkerbell stack services. The updated stack versions PR in `eks-anywhere-build-tooling` is incoming.

*Testing (if applicable):*
Manual tests were run to validate that vlan ids provided in the `hardware.csv` propagated properly into the `hardware.tinkerbell.org` crd.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pkg/providers/tinkerbell/hardware/validator.go [hunk 1]
```diff
@@ -134,6 +136,17 @@ func StaticMachineAssertions() MachineAssertion {
 			}
 		}
 
+		if m.VLANID != "" {
+			i, err := strconv.Atoi(m.VLANID)
+			if err != nil {
+				return errors.New("VLANID: must be an integer")
+			}
+
+			if i < 1 || i > 4094 {
```
Comment (chrisdoherty4) [inline pkg/providers/tinkerbell/hardware/validator.go:145]: Could we throw down a constant name and/or a link to something like https://en.m.wikipedia.org/wiki/VLAN#IEEE_802.1Q so the 4094 is more understandable?

I double took at this and googled it.

File: pkg/providers/tinkerbell/hardware/validator_test.go [hunk 2]
```diff
@@ -206,6 +206,12 @@ func TestStaticMachineAssertions_InvalidMachines(t *testing.T) {
 		"InvalidWithJustDev": func(h *hardware.Machine) {
 			h.Disk = "/dev/"
 		},
+		"InvalidVLAN": func(h *hardware.Machine) {
+			h.VLANID = "0"
+		},
+		"NonIntVLAN": func(h *hardware.Machine) {
```
Comment (chrisdoherty4) [inline pkg/providers/tinkerbell/hardware/validator_test.go:212]: Do we want a maximum bounds rest too?

File: pkg/providers/tinkerbell/hardware/validator.go (hunk 1 above)
Comment (jacobweinstock) [reply pkg/providers/tinkerbell/hardware/validator.go:145]: for sure, will add.

File: pkg/providers/tinkerbell/hardware/validator_test.go (hunk 2 above)
Comment (jacobweinstock) [reply pkg/providers/tinkerbell/hardware/validator_test.go:212]: sounds good, will add.

File: pkg/providers/tinkerbell/hardware/validator.go (hunk 1 above)
Comment (jacobweinstock) [reply pkg/providers/tinkerbell/hardware/validator.go]: added

File: pkg/providers/tinkerbell/hardware/validator_test.go (hunk 2 above)
Comment (jacobweinstock) [reply pkg/providers/tinkerbell/hardware/validator_test.go:215]: added

**Review by chrisdoherty4 — APPROVED** [review]

**Pull request merged**